contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,580 @@
+         mountPath: /var/log/
+         policy:
+           logDir:
+             subDir: bar
+             glob: *.log
+    volumes:
+    - name: myLoggingVolume
+      inline:
+        spec:
+          accessModes: [ "ReadWriteOnce" ]
+          storageClassName: local-slow
+          resources:
+            requests:
+              storage: 1Gi
+    - name: myInMemoryLogVolume
+      emptyDir:
+        medium: memory
+        resources:
+          limits:
+            size: 100Mi
+    ```
+
+5. Phippy notices some of her pods are suffering hangs by while writing to their writable layer. Phippy again notices that I/O contention is the root cause and then updates her Pod Spec to use memory backed or persistent volumes for her pods writable layer. Kubelet will instruct the runtimes to overlay the volume with `overlay` policy over the writable layer of the container.
+
+    ```yaml
+    apiVersion: v1
+    kind: pod
+    metadata:
+     name: foo
+    spec:
+     containers:
+     - name: fooc
+       volumeMounts:
+       - name: myWritableLayer
+         policy:
+           overlay:
+             subDir: foo
+     - name: barc
+       volumeMounts:
+       - name: myDurableWritableLayer
+         policy:
+           overlay:
+               subDir: bar
+     volumes:
+     - name: myWritableLayer
+       emptyDir:
+         medium: memory
+         resources:
+           limits:
+             storage: 100Mi
+     - name: myDurableWritableLayer
+       inline:
+         spec:
+           accessModes: [ "ReadWriteOnce" ]
+           storageClassName: local-fast
+           resources:
+             requests:
+               storage: 1Gi
+    ```
+
+### Bob manages a specialized application that needs access to Block level storage
+
+1. The cluster that Bob uses has nodes that contain raw block devices that have not been formatted yet.
+2. The same addon DaemonSet can discover block devices in the same directory as the filesystem mount points and creates corresponding PVs for them with a new `volumeType = block` field.  This field indicates the original volume type upon PV creation.
+
+    ```yaml
+    kind: PersistentVolume
+    apiVersion: v1
+    metadata:
+      name: foo
+      annotations:
+        storage.kubernetes.io/node: node-1
+    spec:
+      volumeType: block 
+      capacity:
+        storage: 100Gi
+      local:
+        path: /var/lib/kubelet/storage-raw-devices/foo
+      accessModes:
+        - ReadWriteOnce
+      persistentVolumeReclaimPolicy: Delete
+      storageClassName: local-fast
+    ```
+
+3. Bob creates a pod with a PVC that requests for block level access and similar to a Stateful Set scenario the scheduler will identify nodes that can satisfy the pods request.  The block devices will not be formatted to allow the application to handle the device using their own methods.
+
+    ```yaml
+    kind: PersistentVolumeClaim
+    apiVersion: v1
+    metadata:
+      name: myclaim
+    spec:
+      volumeType: block
```

Comment:
To make it easy, I feel we should add a sketch of `StorageClass` as well @msau42. Given that we want users to primarily use Storage Classes, this proposal should illustrate that mainly. PVC including a `volumeType` field then becomes an implementation detail.